test.json

```diff
@@ -1,13 +1,15 @@
 {
   "source": "./package.json",
   "scripts": {
-    "run": "esy '@test' x OniUnitTestRunner",
-    "run-ci": "esy '@test' x OniUnitTestRunnerCI",
+    "run": "esy '@test' x OniUnitTestRunner && esy @test inline",
```

Comment:
Unfortunately the `&&` doesn't seem to be handled the same on Windows - it doesn't actually execute `esy @test inline` (so the `esy '@test' run` call passes, when it should fail due to the failing test case).

For CI - we could add an `esy @test inline` to our build steps, to make sure they get run on Windows: https://github.com/onivim/oni2/blob/c5849ea07e56625445f0151b483f58fd556b2c6e/.ci/esy-build-steps.yml#L26

Reply:
Ah, sorry, seems this is a shortcoming of esy in general actually. For the CI this was just temporary to quickly test it, so no big deal there, but it would have been nice to have a single command to run tests. Not sure how to do that now...

Reply:
> but it would have been nice to have a single command to run tests. 

No worries, maybe there is a way to create an executable to run it (or run it as part of `OniUnitTestRunnerCI`: https://github.com/janestreet/ppx_inline_test#building-and-running-the-tests-outside-of-jane-street) - but not sure if it's worth going to that level.

Reply:
Yeah, if you're cool with just doing `esy @test inline && esy @test run` locally. Should be easier to remember once we get more inline tests, which I expect we will.

Reply:
@Et7f3 claims https://github.com/onivim/oni2/pull/1822/commits/9861370cac07525cb17b22481a6ed11c34e6e620 will work on Windows because of esy-bash. Can you test @bryphe?

Reply:
That's what is done in revery esy.install

Reply:
Unfortunately doesn't seem to work - I just get: 
```
error: project is not installed, run `esy install`
```
on Windows.

I think the way `esy` runs scripts in the `package.json` is unfortunately different than the way `esy` runs the `install`/`build` commands.

Reply:
You have to first install the test sandbox and the default sandbox I think. It is the same than `esy b cmd` it fail if sandbox isn't installed.

Reply:
Hmm, good point @Et7f3 - seems I didn't have the default sandbox installed.

However, after running `esy install`, I'm getting this now with `esy @test run`:
```
/cygdrive/c/Users/bryphe/AppData/Roaming/npm/esy: line 1: /node_modules/esy/_build/default/bin/esy.exe: No such file or directory
```

Reply:
`esy @test inline` runs `esy b dune runtest`, so could just be that too. Runnning `esy '@test ' b dune runtest` instead shouldn't be a problem though.

Reply:
Esy doesn't seem to add itself in the path...
Try:
"bash -c \"dune runtest && dune exec ./test/OniUnitTestRunner\""

Reply:
GitHub removed my backslash 🤨

Reply:
`dune runtest` needs to be run in the build environment though, so maybe `"esy b bash -c \"dune runtest && dune exec ./test/OniUnitTestRunner\""` then?

Reply:
Yup it seem better.

Reply:
Hmm, the inline tests work fine then, but the rely tests fails:

```
Running 1 test suite   
 RUNS  CLIRunning executable: /home/glennsl/dev/remote/glennsl/oni2/Oni2
--version
Got output: ||
Status: WEXITED: 127
Expected output to contain: Onivim 2
But received: 
Running executable: /home/glennsl/dev/remote/glennsl/oni2/Oni2
--list-extensions
--extensions-dir=/home/glennsl/dev/remote/glennsl/oni2/development_extensions
Got output: ||
Status: WEXITED: 127
Expected output to contain: dev-extension
But received: 
Running executable: /home/glennsl/dev/remote/glennsl/oni2/Oni2_editor
-psn_0_989382
--list-extensions
Got output: ||
Status: WEXITED: 127
Expected status to be: WEXITED: 0
But received: WEXITED: 127
 FAIL  CLI
  • CLI › launcher › --version should show output

    Exception (Failure "Output incorrect")

  • CLI › launcher › --list-extensions should show extensions

    Exception (Failure "Output incorrect")

  • CLI › editor › rogue -psn argument shouldn't cause a failure

    Exception (Failure "Exit code incorrect")


Test Suites: 1 failed, 0 passed, 1 total
Tests:       3 failed, 0 passed, 3 total
Time:        0.007s
```

Reply:
Hmm, looks like something strange with the working directory in the `dune exec` case - this path can't be right: `/home/glennsl/dev/remote/glennsl/oni2/Oni2`. 

`dune exec` must be changing how we find the executing directory for the test cases here: https://github.com/onivim/oni2/blob/b537fac3f612f621378900d53df3d45335f04a72/test/Cli/TestRunner.re#L16

Path of least resistance might be to just have the two separate commands for now

Reply:
Yeah, did that for now. They really need to be run in different environments it seems.